cider-dap/src/demo_client.rs

```diff
@@ -0,0 +1,73 @@
+use owo_colors::OwoColorize;
+use std::io::{self, BufRead, BufReader, Read, Write};
+use std::net::TcpStream;
+
+pub(crate) fn send_request(
+    stream: &mut TcpStream,
+    request: &str,
+) -> std::io::Result<String> {
+    let content_length = request.len();
+
+    let formatted_request =
+        format!("Content-Length: {}\r\n\r\n{}", content_length, request);
+
+    println!("{} ", formatted_request.magenta().bold());
+    stream.write_all(formatted_request.as_bytes())?;
+    eprint!("finished writing");
+
+    let mut response = String::new();
+    let mut buf = vec![0u8; 1024];
+    dbg!(stream.peek(&mut buf).unwrap());
+
+    // Read the header line (first line)
+    let mut reader = BufReader::new(stream);
+    let mut header_line = String::new();
+    reader.read_line(&mut header_line)?;
+
+    println!("Header line: {}", header_line.green().bold());
+
+    // Read the empty line (second line)
+    let mut empty_line = String::new();
+    reader.read_line(&mut empty_line)?;
+
+    // Read the specified bytes of the message
+    let mut message_buf = Vec::new();
+    reader.read_until(b'\n', &mut message_buf).unwrap();
```

Comment:
"I gave the message length approach a try, but unfortunately, it didn't work out for me. I suppose I could have spent more time debugging it, but then I thought of trying this different approach instead, and it worked, I guess. Now, I'm wondering if I should replace it with the message.length approach instead. What do you think? Thank you so much for all your input and suggestions!

Reply:
> surprised this works and wondering if you're getting a partial message since this is just reading a single line. Was it the case that you couldn't get the message length approach to work?

I gave the message length approach a try, but unfortunately, it didn't work out for me. I suppose I could have spent more time debugging it, but then I thought of trying this different approach instead, and it worked, I guess. Now, I'm wondering if I should replace it with the message.length approach instead. What do you think? Thank you so much for all your input and suggestions!

Reply:
If it works, I'd say let's just leave it since the demo client was really just to confirm we had the basics working before trying to attach the server through vscode